ebean-api/src/main/java/io/ebean/bean/EntityBeanIntercept.java

```diff
@@ -649,7 +653,7 @@ public void setNewBeanForUpdate() {
   public void addDirtyPropertyNames(Set<String> props, String prefix) {
     int len = getPropertyLength();
     for (int i = 0; i < len; i++) {
-      if ((flags[i] & FLAG_CHANGED_PROP) != 0) {
+      if ((flags[i] & FLAG_CHANGED_PROP) != 0 || isChangedByHash(i)) {
```

Comment:
CHECKME: should we set the `FLAG_CHANGED_PROP` when we detect a change by hash (would save some json conversions)